src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+
+        var newServiceAccountsTotal = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts.Value + serviceAccountAdjustment
+            : serviceAccountAdjustment;
+
+        if (plan.BaseServiceAccount > newServiceAccountsTotal)
+        {
+            throw new BadRequestException($"Plan has a minimum of {plan.BaseServiceAccount} Service Account.");
+        }
+
+        if (newServiceAccountsTotal <= 0)
+        {
+            throw new BadRequestException("You must have at least 1 Service Account.");
+        }
+
+        var additionalServiceAccounts = newServiceAccountsTotal - plan.BaseServiceAccount;
+
+        if (plan.MaxAdditionalServiceAccount.HasValue && additionalServiceAccounts > plan.MaxAdditionalServiceAccount.Value)
+        {
+            throw new BadRequestException($"Organization plan allows a maximum of " +
+                                          $"{plan.MaxAdditionalServiceAccount.Value} additional Service Account.");
+        }
+
+        if (!organization.SmServiceAccounts.HasValue || organization.SmServiceAccounts.Value > newServiceAccountsTotal)
+        {
+            var occupiedServiceAccounts = await _serviceAccountRepository.GetServiceAccountCountByOrganizationIdAsync(organization.Id);
+            if (occupiedServiceAccounts > newServiceAccountsTotal)
+            {
+                throw new BadRequestException($"Your organization currently has {occupiedServiceAccounts} seats filled. " +
+                                              $"Your new plan only has ({occupiedServiceAccounts}) seats. Remove some users.");
+            }
+        }
+
+        var paymentIntentClientSecret = await _paymentService.AdjustServiceAccountsAsync(organization, plan, additionalServiceAccounts.GetValueOrDefault());
+
+        await _referenceEventService.RaiseEventAsync(
+            new ReferenceEvent(ReferenceEventType.AdjustServiceAccounts, organization, _currentContext)
+            {
+                Id = organization.Id,
+                PlanName = plan.Name,
+                PlanType = plan.Type,
+                ServiceAccounts = newServiceAccountsTotal,
+                PreviousServiceAccounts = organization.SmServiceAccounts
+            });
+
+        organization.SmServiceAccounts = newServiceAccountsTotal;
+
+        return paymentIntentClientSecret;
+    }
+
+    private void UpdateSeatsAutoscaling(Organization organization, int? maxAutoscaleSeats)
+    {
+        if (maxAutoscaleSeats.HasValue && organization.SmSeats.HasValue &&
+            maxAutoscaleSeats.Value < organization.SmSeats.Value)
+        {
+            throw new BadRequestException($"Cannot set max Secrets Manager seat autoscaling below current Secrets Manager seat count.");
+        }
+
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
+
+        if (plan == null)
+        {
+            throw new BadRequestException("Existing plan not found.");
+        }
```

Comment:
We do this in every private method. Suggestion: do it once in the public method and then pass it in.

Reply:
sure